src/utils/convertIterableIteratorToAsync.ts

```diff
@@ -3,8 +3,7 @@ export function convertIterableIteratorToAsync<T>(
 ): AsyncIterableIterator<T> {
   return {
     async next() {
-      const result = iterator.next();
-      return Promise.resolve(result);
+      return iterator.next();
```

Comment:
I think `return()` and `throw()` also need to return a Promise.
Also, `next()`, `return()`, and `throw()` can take arguments, so I thought it would be good to support that.

Reply:
Alternatively, if you want to avoid using `return()`, `throw()`, or setting arguments, consider using `satisfies`.

Reply:
@nagasawaryoya 
I have decided not to implement the return() and throw() methods in this PR. According to the ECMAScript specification, these methods are optional. Given the current scope of the implementation and my focus on maintaining simplicity, I believe it's unnecessary to include these methods at this time, especially since there are no complex resource management or error-handling requirements in this context.

Should the need for these methods arise in the future, I am open to revisiting their implementation. However, I have chosen to keep the implementation straightforward by omitting them.